packages/atoms/connect/ConnectButton.tsx

```diff
@@ -0,0 +1,40 @@
+import { useState } from "react";
+
+import type { AtomsGlobalConfigProps } from "../types";
+import type { ConnectButtonProps } from "../types";
+
+// TODO: add default styling
+// also a cssClassAssembler function for custom styling
+
+// we accept two props: buttonText & onButtonClick
+// buttonText is for displaying buttonText
+// onButtonClick is a a method to handle onClick for button
+// it accepts another object which contains a callback in case of any error
+
+export function ConnectButton({ buttonText, onButtonClick }: ConnectButtonProps & AtomsGlobalConfigProps) {
+  const [isProcessing, setIsProcessing] = useState<boolean>(false);
+  const [errMsg, setErrMsg] = useState<string>("");
+
+  function handleSubmit(e: React.MouseEvent<HTMLButtonElement, MouseEvent>) {
+    e.preventDefault();
+
+    setIsProcessing(true);
+    onButtonClick({
+      cb: (err) => {
+        setIsProcessing(false);
+        if (err?.error.message) {
+          setErrMsg(err.error.message);
+        }
+      },
+    });
+  }
+
+  return (
+    <div>
+      <button className="" onClick={(event) => handleSubmit(event)} disabled={!isProcessing} type="button">
+        {buttonText || "Install App"}
```

Comment:
do we wanna do i18n for Cal Atoms?

Reply:
Might have to ask @sean-brydon if handling translations inside our component is better or if we handle those outside? Imo we can pass in a translated string directly as a prop, eg `buttonText={t("Log in")}` and that should work just fine.

Reply:
IMHO button should be able to accept any children -> allowing us to customise it from where it is used. 

Translations inside the connectbutton is a bit weird - im not 100% sure how this would work unless we bundle translations with it

Reply:
for i18n here we would need to only bundle the translations related to an atom inside the atom to avoid providing unnecessary translations to users who only use few atoms. 

regarding accepting a ReactNode as a children is a good option to allow more control for the devs using our components, I think this should be on top of using basic props like label, errorMsg,... for less advanced usage

Reply:
@PeerRich @sean-brydon what do you think about this. Should we just do translations now or leave it for later?